app/controllers/api/network_routers_controller.rb

```diff
@@ -1,4 +1,18 @@
 module Api
   class NetworkRoutersController < BaseController
+    def delete_resource_network_routers(type, id, _data = {})
+      network_router = resource_search(id, type, collection_class(type))
+      task_id = network_router.delete_network_router_queue(User.current_user.id)
+      action_result(true, "Deleting #{network_router_ident(network_router)}", :task_id => task_id)
+    rescue => err
+      action_result(false, err.to_s)
+    end
+    alias network_routers_delete_resource delete_resource_network_routers
```

Comment:
why the alias ?  also since this is a primary collection the method name itself should just be delete_resource. Thanks.